main.py

```diff
@@ -829,79 +830,159 @@ def handle_link_down(self, event):
         """Change circuit when link is down or under_mantenance."""
         link = event.content["link"]
         log.info("Event handle_link_down %s", link)
-        switch_flows = {}
-        evcs_with_failover = []
-        evcs_normal = []
-        check_failover = []
-        failover_event_contents = {}
 
-        for evc in self.get_evcs_by_svc_level():
-            with evc.lock:
-                if evc.is_affected_by_link(link):
-                    evc.affected_by_link_at = event.timestamp
-                    # if there is no failover path, handles link down the
-                    # tradditional way
-                    if (
+        with ExitStack() as exit_stack:
+            exit_stack.enter_context(self._lock)
+            swap_to_failover = list[EVC]()
+            redeploy = list[EVC]()
+            clear_failover = list[EVC]()
+            clear_old_path = list[EVC]()
+            evcs_to_update = list[EVC]()
+
+            for evc in self.get_evcs_by_svc_level():
+                with ExitStack() as sub_stack:
+                    sub_stack.enter_context(evc.lock)
+                    if all((
+                        evc.is_affected_by_link(link),
+                        evc.failover_path,
+                        not evc.is_failover_path_affected_by_link(link)
+                    )):
+                        swap_to_failover.append(evc)
+                    elif all((
+                        evc.is_affected_by_link(link),
                         not evc.failover_path or
                         evc.is_failover_path_affected_by_link(link)
-                    ):
-                        evcs_normal.append(evc)
+                    )):
+                        redeploy.append(evc)
```

Comment:
if `evc.is_affected_by_link(link)` is True 
and `evc.is_failover_path_affected_by_link(link)` is True

failover_path needs to be cleaned too. Otherwise this can end up keeping a failover_path that isn't UP. Can you review this conditional and test it?

Reply:
Well, that's what the old logic was saying to do here. But I do see your point. Looking at the original code, it wouldn't ever tear down the failover path. Just tear down, then setup the current path and try to deploy a failover path if it doesn't already exist.

Reply:
Yes, that was another adjacent bug in the logic. It's the [item 1.2](https://github.com/kytos-ng/mef_eline/issues/517#issuecomment-2511372895). It would leave `failover_path` not UP (with a link down in the path), and then a subsequent link_down in the `current_path` would switchover to this `failover_path` that isn't UP